contrib/guix/guix-build

```diff
@@ -239,7 +239,7 @@ SOURCE_DATE_EPOCH="${SOURCE_DATE_EPOCH:-$(git -c log.showSignature=false log --f
 time-machine() {
     # shellcheck disable=SC2086
     guix time-machine --url=https://git.savannah.gnu.org/git/guix.git \
-                      --commit=aa34d4d28dfe25ba47d5800d05000fb7221788c0 \
+                      --commit=6dffced09ecda024e0884e352778c221ad066fd6 \
```

Comment:
```suggestion
                      --commit=e89f767ce8990f4959616031e3c98fbfe92a008b \
```

makes this PR compatible with `guix` v1.2 packages on [hirsute](https://packages.ubuntu.com/source/hirsute/guix),  [impish](https://packages.ubuntu.com/source/impish/guix) and [bullseye](https://packages.debian.org/bullseye/guix).

Reply:
That commit is not in the 1.4.0 branch.

Reply:
> That commit is not in the 1.4.0 branch.

Right. It's on master after branching `version-1.4.0` off. And what are drawbacks of that?

Reply:
As an alternative we could document that when using a system `guix` package, version >=1.3 is required.

Reply:
Both comments are addressed below.